### PR TITLE
mcs: remove wait api service ready

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref #8477

### What is changed and how does it work?

To support dynamically switching TSO, we must utilize the service discovery to decide who is responsible for serving TSO requests. However, the current implementation will register the TSO service until the API server is ready to serve. So there must be at least one switch when we use ms mode to start the cluster.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/resourcemanager/server/server.go [hunk 1]
```diff
@@ -103,21 +105,49 @@ func (s *Server) SetLogLevel(level string) error {
 
 // Run runs the Resource Manager server.
 func (s *Server) Run() (err error) {
-	skipWaitAPIServiceReady := false
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/mcs/resourcemanager/server/server.go:106]: InitClusterID now will retry until the etcd can provide the service.

File: pkg/mcs/scheduling/server/config/config.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ import (
 	"github.com/tikv/pd/pkg/cache"
 	"github.com/tikv/pd/pkg/core/constant"
 	"github.com/tikv/pd/pkg/core/storelimit"
-	"github.com/tikv/pd/pkg/mcs/utils"
+	mcsconstat "github.com/tikv/pd/pkg/mcs/utils/constant"
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/config/config.go:37]:
```suggestion
	mcsconstant "github.com/tikv/pd/pkg/mcs/utils/constant"
```

File: tests/integrations/mcs/discovery/register_test.go [hunk 3]
```diff
@@ -113,9 +113,7 @@ func (suite *serverRegisterTestSuite) checkServerRegister(serviceName string) {
 }
 
 func (suite *serverRegisterTestSuite) TestServerPrimaryChange() {
-	suite.checkServerPrimaryChange(utils.TSOServiceName, 3)
-	// TODO: uncomment after resource-manager is ready
```
Comment (lhy1024) [inline tests/integrations/mcs/discovery/register_test.go:117]: remove this comment?

File: pkg/mcs/utils/util.go [hunk 4]
```diff
@@ -45,30 +47,26 @@ import (
...
+	retryTimes := 0
+	for {
+		if clusterID, err := etcdutil.GetClusterID(client, constant.ClusterIDPath); err == nil && clusterID != 0 {
 			return clusterID, nil
 		}
 		select {
 		case <-ctx.Done():
 			return 0, err
 		case <-ticker.C:
+			retryTimes++
```
Comment (lhy1024) [inline pkg/mcs/utils/util.go:63]: It will print the log every 250s until it succeeds?

Comment (rleungx) [reply pkg/mcs/utils/util.go:63]: same as previous

File: tests/integrations/mcs/discovery/register_test.go (hunk 3 above)
Comment (rleungx) [reply tests/integrations/mcs/discovery/register_test.go:117]: yes

**Review by lhy1024 — APPROVED** [review]

File: pkg/mcs/resourcemanager/server/server.go [hunk 5]
```diff
@@ -103,18 +109,15 @@ func (s *Server) SetLogLevel(level string) error {
...
-	if !skipWaitAPIServiceReady {
-		if err := utils.WaitAPIServiceReady(s); err != nil {
-			return err
-		}
+	if err = utils.InitClient(s); err != nil {
+		return err
 	}
-	if err := utils.InitClient(s); err != nil {
+
+	// register
```
Comment (okJiang) [inline pkg/mcs/resourcemanager/server/server.go:116]: This is a meaningless comment. We can remove it

File: pkg/mcs/scheduling/server/server.go [hunk 6]
```diff
@@ -140,20 +144,16 @@ func (s *Server) SetLogLevel(level string) error {
...
-		if err := utils.WaitAPIServiceReady(s); err != nil {
-			return err
-		}
+func (s *Server) Run() (err error) {
+	if err = utils.InitClient(s); err != nil {
+		return err
 	}
 
-	if err := utils.InitClient(s); err != nil {
+	// register
```
Comment (okJiang) [inline pkg/mcs/scheduling/server/server.go:152]: ditto

File: pkg/mcs/server/server.go [hunk 7]
```diff
@@ -100,6 +100,11 @@ func (bs *BaseServer) SetETCDClient(etcdClient *clientv3.Client) {
 	bs.etcdClient = etcdClient
 }
 
+// SetETCDClient sets the etcd client.
```
Comment (okJiang) [inline pkg/mcs/server/server.go:103]: typo


```suggestion
// GetETCDClient gets the etcd client.
```

File: pkg/mcs/tso/server/server.go [hunk 8]
```diff
@@ -143,24 +146,21 @@ func (s *Server) SetLogLevel(level string) error {
...
 		log.Error("system time jumps backward", errs.ZapError(errs.ErrIncorrectSystemTime))
 		timeJumpBackCounter.Inc()
 	})
 
-	if err := utils.InitClient(s); err != nil {
+	if err = utils.InitClient(s); err != nil {
 		return err
 	}
+
+	// register
```
Comment (okJiang) [inline pkg/mcs/tso/server/server.go:159]: ditto

File: pkg/mcs/utils/util.go [hunk 9]
```diff
@@ -330,6 +278,41 @@ func StopGRPCServer(s server) {
 	}
 }
 
+func Register(s server, serviceName string) (uint64, *discovery.ServiceRegistryEntry, *discovery.ServiceRegister, error) {
```
Comment (okJiang) [inline pkg/mcs/utils/util.go:281]: Please add some comments for it

File: pkg/mcs/utils/util.go
Comment (okJiang) [inline pkg/mcs/utils/util.go:1]: Most of the content in the file is related to the `server`. Maybe we can rename the file, or later move the server-related parts to `server.go`.

**Review by okJiang — APPROVED** [review]
Please fix the conflict

**Review by JmPotato — APPROVED** [review]

**Review by niubell — APPROVED** [review]

**Pull request merged**